bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -375,92 +383,95 @@ async fn extension_loop_active(
-                                lambda_enhanced_metrics
-                                    .set_init_duration_metric(metrics.duration_ms);
-                            }
-                            TelemetryRecord::PlatformRuntimeDone {
-                                request_id,
-                                status,
-                                metrics,
-                                ..
-                            } => {
-                                if let Some(metrics) = metrics {
-                                    invocation_context_buffer
-                                        .add_runtime_duration(&request_id, metrics.duration_ms);
+                        Event::Telemetry(event) =>
+                            match event.record {
+                                TelemetryRecord::PlatformStart { request_id, .. } => {
+                                    let mut p = invocation_processor.lock().await;
+                                    p.on_platform_start(request_id, event.time);
+                                    drop(p);
+                                }
+                                TelemetryRecord::PlatformInitReport {
+                                    initialization_type,
+                                    phase,
+                                    metrics,
+                                } => {
+                                    debug!("Platform init report for initialization_type: {:?} with phase: {:?} and metrics: {:?}", initialization_type, phase, metrics);
                                     lambda_enhanced_metrics
-                                        .set_runtime_duration_metric(metrics.duration_ms);
+                                        .set_init_duration_metric(metrics.duration_ms);
                                 }
+                                TelemetryRecord::PlatformRuntimeDone {
+                                    request_id,
+                                    status,
+                                    metrics,
+                                    ..
+                                } => {
+                                    let mut p = invocation_processor.lock().await;
+                                    if let Some(metrics) = metrics {
+                                        p.on_platform_runtime_done(
+                                            &request_id,
+                                            metrics.duration_ms,
+                                            config.clone(),
+                                            tags_provider.clone(),
+                                            trace_processor.clone(),
+                                            trace_agent_tx.clone()
+                                        ).await;
+                                        lambda_enhanced_metrics
+                                            .set_runtime_duration_metric(metrics.duration_ms);
+                                    }
```

Comment:
I might want to delegate some of this metrics to the lifecycle processor eventually, maybe?

Reply:
it'd be good to push this out somewhere which can do the work asynchronously, you're blocking the loop right now